framework/components/ecl-accordions/accordions.js

```diff
@@ -0,0 +1,169 @@
+      header.setAttribute('tabindex', -1);
+    });
+    // set active focus
+    headerSet[i].setAttribute('tabindex', 0);
+    headerSet[i].focus();
+  }
+
+  // EVENTS
+  function eventHeaderClick(e) {
+    togglePanel(e.currentTarget);
+  }
+
+  function eventHeaderKeydown(e) {
+    // collect header targets, and their prev/next
+    const currentHeader = e.currentTarget;
+    const isModifierKey = e.metaKey || e.altKey;
+    // get context of accordion container and its children
+    const thisContainer = currentHeader.parentNode;
+    const theseHeaders = queryAll(headerSelector, thisContainer);
+    const currentHeaderIndex = [].indexOf.call(theseHeaders, currentHeader);
+
+    // don't catch key events when ⌘ or Alt modifier is present
+    if (isModifierKey) return;
+
+    // catch enter/space, left/right and up/down arrow key events
+    // if new panel show it, if next/prev move focus
+    switch (e.keyCode) {
+      case 13:
+      case 32:
+        togglePanel(currentHeader);
+        e.preventDefault();
+        break;
+      case 37:
+      case 38: {
+        const previousHeaderIndex = currentHeaderIndex === 0
+          ? theseHeaders.length - 1
+          : currentHeaderIndex - 1;
+        giveHeaderFocus(theseHeaders, previousHeaderIndex);
+        e.preventDefault();
+        break;
+      }
+      case 39:
+      case 40: {
+        const nextHeaderIndex = currentHeaderIndex < theseHeaders.length - 1
+          ? currentHeaderIndex + 1
+          : 0;
+        giveHeaderFocus(theseHeaders, nextHeaderIndex);
+        e.preventDefault();
+        break;
+      }
+      default:
+        break;
+    }
+  }
+
+  // BIND EVENTS
+  function bindAccordionEvents(accordionContainer) {
+    const accordionHeaders = queryAll(headerSelector, accordionContainer);
+    // bind all accordion header click and keydown events
+    accordionHeaders.forEach(accordionHeader => {
+      accordionHeader.addEventListener('click', eventHeaderClick);
+      accordionHeader.addEventListener('keydown', eventHeaderKeydown);
+    });
+  }
+
+  // UNBIND EVENTS
+  function unbindAccordionEvents(accordionContainer) {
+    const accordionHeaders = queryAll(headerSelector, accordionContainer);
+    // unbind all accordion header click and keydown events
+    accordionHeaders.forEach(accordionHeader => {
+      accordionHeader.removeEventListener('click', eventHeaderClick);
+      accordionHeader.removeEventListener('keydown', eventHeaderKeydown);
+    });
+  }
+
+  // DESTROY
+  function destroy() {
+    accordionContainers.forEach(accordionContainer => {
+      unbindAccordionEvents(accordionContainer);
+    });
+  }
+
+  // INIT
+  function init() {
+    if (accordionContainers.length) {
+      accordionContainers.forEach(accordionContainer => {
+        bindAccordionEvents(accordionContainer);
+      });
+    }
+  }
+
+  init();
+
+  // REVEAL API
+  return {
+    init,
+    destroy,
+  };
+};
+
+// module exports
+export default accordions;
```

Comment:
so in the end, accordions do not rely on expandables at all. That's quite a shame, but that's probably easier this way.